### PR TITLE
improve volk_8i_s32f_convert_32f for ARM NEON

### PR DESCRIPTION
**Pull request by dernasherbrezon**
Current implementation is based on array of vectors (float32x4x2_t). I
switched it to normal vector - float32x4_t and got ~80ms improvement on
RaspberryPI 3+

Before:
generic completed in 446.774 ms
neon completed in 434.753 ms

After:
generic completed in 455.331 ms
neon completed in 376.777 ms

### CODE REVIEW

**Comment by jdemel** [comment]
Thanks for your PR @dernasherbrezon !

We do have issues with our CI pipeline for ARM. In other words, due to a policy change at TravisCI, our ARM test don't run at the moment. Thus, it is terribly difficult to verify your PR. I'd like to ask for patience in merging this PR. And again: Thanks for your contribution.

File: kernels/volk/volk_8i_s32f_convert_32f.h [hunk 1]
```diff
@@ -307,41 +307,37 @@ static inline void volk_8i_s32f_convert_32f_neon(float* outputVector,
...
-        outputFloat.val[0] = vmulq_f32(outputFloat.val[0], qiScalar);
-        vst1q_f32(outputVectorPtr, outputFloat.val[0]);
+        outputFloat = vmulq_f32(vcvtq_f32_s32(vmovl_s16(vget_low_s16(higher))), qiScalar);
+        vst1q_f32(outputVectorPtr, outputFloat);
         outputVectorPtr += 4;
 
-        outputFloat.val[1] = vcvtq_f32_s32(vmovl_s16(vget_high_s16(tmp)));
-        outputFloat.val[1] = vmulq_f32(outputFloat.val[1], qiScalar);
-        vst1q_f32(outputVectorPtr, outputFloat.val[1]);
+        outputFloat =
```
Comment (michaelld) [inline kernels/volk/volk_8i_s32f_convert_32f.h:338]: Remove the \n here to make this & the next line just 1 line

**Review by michaelld — CHANGES_REQUESTED** [review]
1 more formatting tweak & then squash the commits & then LGTM

**Comment by michaelld** [comment]
I'll echo @jdemel here: we don't have great ARM 32/64 CI testing at the moment ... I personally don't have a good way to test this architecture at the moment. I'm working to bring up a RPi4, but that's not a priority compared with other work I'm doing, so I'll get there when I get there. That said, the code reads well & I'm willing to trust the PR authors to have done basic diligence with testing for the correct ARCH. So beyond formatting & squashing commits I'm good here. Thanks for your contribution!

File: kernels/volk/volk_8i_s32f_convert_32f.h (hunk 1 above)
Comment (jdemel) [reply kernels/volk/volk_8i_s32f_convert_32f.h:338]: While this looks odd, it passes our code format checks. `clang-format` does that automatically. I suggest to leave it as is.

**Comment by jdemel** [comment]
We just got some aarch64 tests back with Actions. Though, RPi3 is mostly 32bit and these tests are not yet available again. It would be great if you could rebase your PR onto current master. If you're curious, I'd be happy to see help with our CI to test for 32bit ARM as well.

File: kernels/volk/volk_8i_s32f_convert_32f.h (hunk 1 above)
Comment (dernasherbrezon) [reply kernels/volk/volk_8i_s32f_convert_32f.h:338]: Yes, the version without \n didn't pass the formatting checks...

**Review by dernasherbrezon — COMMENTED** [review]
I tested it on RPi3, which is 32bit arm. How can I check aarch64 (arm64?) test results?

**Comment by jdemel** [comment]
> I tested it on RPi3, which is 32bit arm. How can I check aarch64 (arm64?) test results?

I'd assume the easiest way right now would be to rebase your PR on current master. It does include tests for aarch64 that should be triggered as soon as you push your rebased branch to github.

**Comment by dernasherbrezon** [comment]
Guys, looks like some tests are unstable. "qa_volk_16ic_x2_dot_prod_16ic". Do you know how to re-run the build?

**Comment by michaelld** [comment]
I forced rerunning all CI jobs. That also happens when someone pushes (force or regularly) updates to the branch trying to be merged for this PR.

**Comment by dernasherbrezon** [comment]
Still failing. What should I do? Re-create branch/pull request?

**Comment by dernasherbrezon** [comment]
@michaelld So what we are going to do? Should I re-create branch and pull request or we try to re-run the CI jobs one more time?

**Comment by jdemel** [comment]
All ARM related tests pass. Unfortunately if one of the test in a build matrix fail, all others are cancelled. In your case these are x86 tests where one of our CI tests fails. I suggest to ignore them here.

**Comment by jdemel** [comment]
I merge this PR now. Comments have been addressed. Relevant CI passes.

**Pull request merged**